nstar/src/main.rs

```diff
@@ -12,306 +12,198 @@
                 } else {
-                    "stop".to_string()
+                    // TODO somehow handle multiple errors
+                    return Ok(Response::Err(api::model::Error::Process(error)));
                 }
             }
-            Command::Shutdown => "shutdown".to_string(),
-            Command::Install { npk, repository } => {
-                format!("install {} {}", npk.display(), repository)
-            }
-            Command::Uninstall { name, version } => format!("uninstall {} {}", name, version),
         }
-    }
-}
-
-const HELP: &str = r"containers:                   List installed containers
-repositories:                 List available repositories
-shutdown:                     Stop the northstar runtime
-start <name>:                 Start application
-stop <name>:                  Stop application
-install <npk-file> <repo-id>: Install npk into repository
-uninstall <name> <version>:   Unstall npk";
-
-async fn process<W: std::io::Write>(
-    client: &mut Client,
-    output: &mut W,
-    input: &str,
-) -> Result<Option<String>> {
-    let mut split = input.split_whitespace();
-    if let Some(cmd) = split.next() {
-        match cmd {
-            "help" | "h" => {
-                writeln!(output, "{}", HELP)?;
-            }
-            "containers" | "ls" => {
-                let containers = client.containers().await?;
-                pretty::containers(output, &containers)?;
-            }
-            "repositories" => {
-                let repositories = client.repositories().await?;
-                pretty::repositories(output, &repositories)?;
-            }
-            "start" => {
-                let mut containers = client.containers().await?;
-                let containers = containers
-                    .drain(..)
-                    .filter(|c| c.manifest.init.is_some()) // Filter resource containers
-                    .filter(|c| c.process.is_none()) // Filter started containers
-                    .map(|c| c.manifest.name)
-                    .collect::<Vec<_>>();
-                if let Some(n) = split.next() {
-                    // Exact match
-                    if containers.iter().any(|c| c == n) {
-                        client.start(n).await?;
-                    } else {
-                        let re = regex::Regex::new(n)?;
-                        for name in containers.iter().filter(|c| re.is_match(&c)) {
-                            client.start(&name).await?;
+        Northstar::Stop { name } => {
+            if let Some(name) = name {
+                Request::Stop(name.to_owned())
+            } else {
+                use std::fmt::Write;
+                let mut error = String::new();
+                for name in running_containers(client).await? {
+                    match client.stop(&name).await {
+                        Err(api::client::Error::Api(e)) => {
+                            writeln!(&mut error, "{}: {:?}", name, e)?;
```

Comment:
Tricky. Returning an Error if one start failed if hiding the positive results. Not sure what to do best here but probably abandon the possibility to start multiple things with a single command.